plugin/storage/es/spanstore/writer.go

```diff
@@ -0,0 +1,205 @@
+        "spanID": 		{ "type": "keyword", "ignore_above": 256 },
+        "operationName": 	{ "type": "keyword", "ignore_above": 256 },
+        "startTime": 		{ "type": "long" },
+        "duration": 		{ "type": "long" },
+        "flags": 		{ "type": "integer" },
+        "logs": {
+          "properties": {
+            "timestamp": 	{ "type": "long" },
+            "tags": {
+              "type": 		"nested",
+              "dynamic":  	false,
+              "properties": {
+                "key": 		{ "type": "keyword", "ignore_above": 256 },
+                "value": 	{ "type": "keyword", "ignore_above": 256 },
+                "tagType": 	{ "type": "keyword", "ignore_above": 256 }
+              }
+            }
+          }
+        },
+        "process": {
+          "properties": {
+            "serviceName": 	{ "type": "keyword", "ignore_above": 256 },
+            "tags": {
+              "type": 		"nested",
+              "dynamic":  	false,
+              "properties": {
+                "key": 		{ "type": "keyword", "ignore_above": 256 },
+                "value": 	{ "type": "keyword", "ignore_above": 256 },
+                "tagType": 	{ "type": "keyword", "ignore_above": 256 }
+              }
+            }
+          }
+        },
+        "references": {
+          "type":   		"nested",
+          "dynamic": 		false,
+          "properties": {
+            "refType": 		{ "type": "keyword", "ignore_above": 256 },
+            "traceID": 		{ "type": "keyword", "ignore_above": 256 },
+            "spanID": 		{ "type": "keyword", "ignore_above": 256 },
+          }
+        },
+        "tags": {
+          "type": 		"nested",
+          "dynamic": false,
+          "properties": {
+            "key": 		{ "type": "keyword", "ignore_above": 256 },
+            "value": 		{ "type": "keyword", "ignore_above": 256 },
+            "tagType": 		{ "type": "keyword", "ignore_above": 256 }
+          }
+        }
+      }
+    },
+    "service": {
+      "properties": {
+        "serviceName": 		{ "type": "keyword", "ignore_above": 256 },
+        "operationName": 	{ "type": "keyword", "ignore_above": 256 }
+      }
+    }
+  }
+}`
+
+// SpanWriter is a wrapper around elastic.Client
+type SpanWriter struct {
+	client es.Client
+	logger *zap.Logger
+}
+
+// Service is the JSON struct for service:operation documents in ElasticSearch
+type Service struct {
+	serviceName   string
+	operationName string
+}
+
+// NewSpanWriter creates a new SpanWriter for use
+func NewSpanWriter(
+	client es.Client,
+	logger *zap.Logger,
+) *SpanWriter {
+	return &SpanWriter{
+		client: client,
+		logger: logger,
+	}
+}
+
+// WriteSpan writes a span and its corresponding service:operation in ElasticSearch
+func (s *SpanWriter) WriteSpan(span *model.Span) error {
+	// Convert model.Span into json.Span
+	jsonSpan := json.FromDomainEmbedProcess(span)
+
+	ctx := context.Background()
+
+	today := time.Now().Format("2006-01-02")
```

Comment:
does zipkin use the starttime of the actual span or the time.now()? How do they handle the case where the span starts in one index and finishes in the next index?

Also, this can be it's own little function getIndexName()